tests/validate/aie_pl_test/src/host.cpp

```diff
@@ -198,13 +134,203 @@ main(int argc, char* argv[])
+    pm_bo.sync(XCL_BO_SYNC_BO_TO_DEVICE, (num_pm + 1) * sizeof(uint32_t),
+               /*OFFSET=*/0);
+    std::cout << "sync pm buffer complete" << std::endl;
+    // start sender_receiver kernels
+    auto sender_receiver_r1 = xrt::run(sender_receiver_k1);
+    sender_receiver_r1.set_arg(0, num_iter);
+    sender_receiver_r1.set_arg(1, num_sample);
+    sender_receiver_r1.set_arg(2, in_bo1);
+    sender_receiver_r1.set_arg(3, out_bo1);
+    sender_receiver_r1.start();
+    std::cout << " start sender-receiver kernel" << std::endl;
+
+    // start pl controller
+    auto controller_r1 = xrt::run(controller_k1);
+    int ctrl_pkt_id = 0;
+    controller_r1.set_arg(2, ctrl_pkt_id);
+    controller_r1.set_arg(3, pm_bo);
+    controller_r1.start();
+    std::cout << "start pl controller kernel" << std::endl;
+    // start input kernels
+
+    controller_r1.wait();
+    // sync output memory
+    out_bo1.sync(XCL_BO_SYNC_BO_FROM_DEVICE, mem_size, /*OFFSET=*/0);
+    // post-processing data;
+    int i;
+    for (i = 0; i < mem_size / sizeof(int); i++) {
+      if (*(host_out1 + i) != *(host_in1 + i) + 1) {
+	match = true;
+	std::cout << "host_out1[" << i << "]=" << host_out1[i] << std::endl;
+      }
+    }
+
+    return match;
+}
+
+int
+main(int argc, char* argv[])
+{
+  std::string b_file;
+
+    // Option Variables
+    std::string test_path;
+    std::string dev_id = "0";
+    bool flag_s = false;
+
+    // -- Retrieve and parse the subcommand options
+    // -----------------------------
+
+    po::options_description desc("Available Options");
+    desc.add_options()(
+        "path,p", boost::program_options::value<decltype(test_path)>(&test_path)
+                      ->required(),
+        "Platform test path")(
+        "device,d", boost::program_options::value<decltype(dev_id)>(&dev_id),
+        "Device ID")("supported,s",
+                     boost::program_options::bool_switch(&flag_s),
+                     "Supported")("help,h", "Prints this help menu.");
+
+    po::positional_options_description positionals;
+
+    // Parse sub-command ...
+    po::variables_map vm;
+    try {
+        po::store(po::command_line_parser(argc, argv)
+                      .options(desc)
+                      .positional(positionals)
+                      .run(),
+                  vm);
+        // Print out the help menu
+        if (vm.count("help") != 0) {
+            std::cout << desc << "\n";
+            return EXIT_SUCCESS;
+        }
+
+        // Validate the options
+        po::notify(vm); // Can throw
+    } catch (po::error& e) {
+        std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+        std::cout << desc << "\n";
+        return EXIT_FAILURE;
+    }
+
+    std::string aie_control_file = "/aie_control_config.json";
+    std::string aie_control = test_path + aie_control_file;
+    std::ifstream aiefile(aie_control);
+    if (!aiefile.good()) {
+        std::cout << "\nNOT SUPPORTED" << std::endl;
```

Comment:
Should this be an error and sent out to std::cerr

Reply:
This is how all the other validate tests are handled.  
I think xbutil will take the string output and decide if the test can be supported or not.